src/waveform/waveform.h

```diff
@@ -143,8 +151,8 @@ class Waveform {
 
     // If stored in the database, the ID of the waveform.
     int m_id;
-    // AnalysisDAO needs to be able to set the waveform as clean.
-    mutable bool m_bDirty;
+    // mutable since AnalysisDAO needs to be able to set the waveform as saved.
+    mutable SaveState m_saveState;
```

Comment:
These two variables are used for different things, and are not exchangeable.
A complete waveform (i.e. one where m_complete is equal to m_datasize), may or may not need to be saved, depending if its contents have been saved already. m_complete could be used to tell that a waveform is "Changing", but would still require checking the state